src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.NonAndroid.cs

```diff
@@ -393,16 +401,21 @@ private static bool TryGetLocalTzFile([NotNullWhen(true)] out byte[]? rawData, [
             id = null;
             string? tzVariable = GetTzEnvironmentVariable();
 
-            // If the env var is null, use the localtime file
+            // If the env var is null, on iOS/tvOS, grab the default tz from the device.
+            // On all other platforms, use the localtime file.
             if (tzVariable == null)
             {
+#if TARGET_IOS || TARGET_TVOS
+                tzVariable = Interop.Sys.GetDefaultTimeZone();
+#else
                 return
                     TryLoadTzFile("/etc/localtime", ref rawData, ref id) ||
                     TryLoadTzFile(Path.Combine(GetTimeZoneDirectory(), "localtime"), ref rawData, ref id);
+#endif
             }
 
             // If it's empty, use UTC (TryGetLocalTzFile() should return false).
-            if (tzVariable.Length == 0)
+            if (tzVariable!.Length == 0)
```

Comment:
now that `pal_datetime.m` returns `NULL` when `NSTimeZone localTimeZone`'s name is `nil`, this will need to check for `null`.

```suggestion
            if (string.IsNullOrEmpty(tzVariable))
```